src/archivist/recorder/repositories/git/index.test.js

```diff
@@ -712,4 +712,188 @@ describe('GitRepository', () => {
+        content: 'content',
+        message: 'Start tracking Service Terms',
+        date: new Date('2023-02-28T01:00:00.000Z'),
+      },
+      deprecatedRefilter: {
+        path: 'service/terms-deprecated.md',
+        content: 'content refiltered',
+        message: 'Refilter Service Terms',
+        date: new Date('2023-02-28T02:00:00.000Z'),
+      },
+      deprecatedUpdate: {
+        path: 'service/terms-deprecated.md',
+        content: 'content updated',
+        message: 'Update Service Terms',
+        date: new Date('2023-02-28T03:00:00.000Z'),
+      },
+      currentFirstRecord: {
+        path: 'service/terms-current.md',
+        content: 'content',
+        message: 'Initial record of Service Terms',
+        date: new Date('2023-02-28T04:00:00.000Z'),
+      },
+      currentExtractOnly: {
+        path: 'service/terms-current.md',
+        content: 'content extract only',
+        message: 'Apply technical or declaration upgrade on Service Terms',
+        date: new Date('2023-02-28T05:00:00.000Z'),
+      },
+      currentUpdate: {
+        path: 'service/terms-current.md',
+        content: 'content updated',
+        message: 'Record new changes of Service Terms',
+        date: new Date('2023-02-28T06:00:00.000Z'),
+      },
+    };
+
+    before(async function () {
+      this.timeout(5000);
+      git = new Git({
+        path: RECORDER_PATH,
+        author: {
+          name: config.get('recorder.versions.storage.git.author.name'),
+          email: config.get('recorder.versions.storage.git.author.email'),
+        },
+      });
+
+      await git.initialize();
+      subject = new GitRepository({
+        ...config.get('recorder.versions.storage.git'),
+        path: RECORDER_PATH,
+      });
+
+      await subject.initialize();
+
+      /* eslint-disable no-await-in-loop */
+      for (const commit of Object.values(commits)) {
+        const { path: relativeFilePath, date, content, message } = commit;
+        const filePath = path.join(RECORDER_PATH, relativeFilePath);
+
+        await GitRepository.writeFile({ filePath, content });
+
+        const sha = await subject.commit({ filePath, message, date });
+
+        commit.id = sha;
+        expectedIds.push(sha);
+        expectedDates.push(date);
+      }
+      /* eslint-enable no-await-in-loop */
+    });
+
+    after(async () => subject.removeAll());
+
+    describe('Records attributes', () => {
+      describe('#isExtractOnly', () => {
+        context('records with deprecated message', () => {
+          it('returns the proper value', async () => {
+            expect((await subject.findById(commits.deprecatedRefilter.id)).isExtractOnly).to.be.true;
+          });
+
+          it('returns the proper value', async () => {
+            expect((await subject.findById(commits.deprecatedFirstRecord.id)).isExtractOnly).to.be.false;
+          });
+        });
+
+        context('record with current message', () => {
```

Comment:
This context conflicts with the broader context: we are actually testing both the “current” and “deprecated” versions. I recommend to isolate the compatibility tests.